pyiceberg/table/metadata.py

```diff
@@ -121,7 +122,7 @@ def check_sort_orders(table_metadata: TableMetadata) -> TableMetadata:
 
 def construct_refs(table_metadata: TableMetadata) -> TableMetadata:
     """Set the main branch if missing."""
-    if table_metadata.current_snapshot_id is not None:
+    if table_metadata.current_snapshot_id is not None and table_metadata.current_snapshot_id != -1:
```

Comment:
Is this -1 check still necessary? `construct_refs` is an after validator. At this point, `cleanup_snapshot_id` should already turn `current_snapshot_id=-1` to `None`.

Reply:
You are right! I'll submit a fix thank you!